python/lsst/ip/diffim/dipoleFitTask.py

```diff
@@ -0,0 +1,921 @@
+
+    maxFluxRatio = pexConfig.Field(
+        dtype=float, default=0.65,
+        doc="Maximum flux ratio in either lobe to be considered a dipole")
+
+    # Choose a maxChi2DoF corresponding to a significance level of at most 0.05
+    # (note this is actually a significance not a chi2 number)
+    maxChi2DoF = pexConfig.Field(
+        dtype=float, default=0.05,
+        doc="Maximum Chi2/DoF significance of fit to be considered a dipole.")
+
+    verbose = pexConfig.Field(
+        dtype=bool, default=False,
+        doc="Be verbose; this is slow")
+
+
+class DipoleFitTask(measBase.SingleFrameMeasurementTask):
+    """This is a subclass of SingleFrameMeasurementTask which can accept
+    three input images in its run() method. Because it subclasses
+    SingleFrameMeasurementTask, and calls
+    SingleFrameMeasurementTask.run() from its run() method, it still
+    can be used identically to a standard SingleFrameMeasurementTask.
+    """
+
+    ConfigClass = DipoleFitPluginConfig
+    _DefaultName = "ip_diffim_DipoleFit"
+
+    def __init__(self, schema, algMetadata=None, dpFitPluginConfig=None, **kwds):
+
+        measBase.SingleFrameMeasurementTask.__init__(self, schema, algMetadata, **kwds)
+
+        if dpFitPluginConfig is None:
+            dpFitPluginConfig = DipoleFitPluginConfig()
+        self.dipoleFitter = DipoleFitPlugin(dpFitPluginConfig, name=self._DefaultName,
+                                            schema=schema, metadata=algMetadata)
+
+    def run(self, sources, exposure, posImage=None, negImage=None, **kwds):
+        """!Run dipole measurement and classification
+
+        @param sources       diaSources that will be measured using dipole measurement
+        @param exposure      Exposure on which the diaSources were detected
+        @param posImage      "Positive" exposure from which the template was subtracted
+        @param negImage      "Negative" exposure which was subtracted from the posImage
+        @param **kwds        Sent to SingleFrameMeasurementTask
+        """
+
+        measBase.SingleFrameMeasurementTask.run(self, sources, exposure, **kwds)
+
+        if not sources:
+            return
+
+        for source in sources:
+            self.dipoleFitter.measure(source, exposure, posImage, negImage)
+
+
+@measBase.register("ip_diffim_DipoleFit")
+class DipoleFitPlugin(measBase.SingleFramePlugin):
+    """Subclass of SingleFramePlugin which can accept three input images
+    in its measure() method and fits dipoles to all merged (two-peak)
+    footprints in a diffim. If provided, it includes data from the
+    pre-subtraction posImage (science image) and optionally negImage
+    (template image) to constrain the fit. The meat of the fitting
+    routines are in the class DipoleFitAlgorithm.
+
+    The motivation behind this plugin and the necessity for including more than
+    one exposure are documented in DMTN-007 (http://dmtn-007.readthedocs.org).
```

Comment:
Did you verify that this format resulted in a link? I worry because the comment starts with """ instead of """!

It is easy to check: build the package and then open doc/html/index.html in your browser